vault/expiration.go

```diff
@@ -225,7 +264,7 @@ func (r *revocationJob) OnFailure(err error) {
 			return
 		}
 
-		r.m.markLeaseAsZombie(r.nsCtx, le, errors.New("lease has consumed all retry attempts"))
+		r.m.markLeaseAsZombie(r.nsCtx, le, errOutOfRetries)
```

Comment:
We are storing the entire lease entry in memory here but with pending & nonexpiring we only store a subset of information to save memory. Should we be calling `inMemoryLeaseInfo` in `markLeaseAsZombie`?

Reply:
yes, good call ddd285b41a